src/plots/gl3d/scene.js

```diff
@@ -31,6 +30,32 @@ var createAxesOptions = require('./layout/convert');
 var createSpikeOptions = require('./layout/spikes');
 var computeTickMarks = require('./layout/tick_marks');
 
+var isMobile = require('is-mobile');
+var tablet = isTablet();
+
+function isTablet() {
+    var navigator = window.navigator;
```

Comment:
Does `is-mobile` setup a guard for cases when `window.navigator` is undefined?

Reply:
Good call.

Reply:
I don't see any check there concerning `navigator`.

Reply:
OK, can you add one just to be sure?

Reply:
(I wouldn't someone running plotly.js in some sort of weird environment have their app break in a patch release)

Reply:
Actually it is added in https://github.com/plotly/plotly.js/pull/4546/commits/436fca46dad2226913b07f3ab0d01b04cb0ad03c here:
```js
if(!ua && typeof navigator !== 'undefined') ua = navigator.userAgent;
```

Reply:
Sure, we can still can breakage on this line below:

```js
navigator.maxTouchPoints > 1
```

if `window.navigator` isn't defined.

Reply:
Addressed in https://github.com/plotly/plotly.js/pull/4546/commits/aa28ebafd53a0fc5cb34a54b5f0718cb0aad4f43.